File: buybeats/src/main/java/com/ibm/buybeats/service/UserService.java
package com.ibm.buybeats.service;

import com.ibm.buybeats.bean.Login;
import com.ibm.buybeats.entity.Address;
import com.ibm.buybeats.entity.CardDetails;
import com.ibm.buybeats.entity.User;
import com.ibm.buybeats.exception.EmailAlreadyExistsException;
import com.ibm.buybeats.exception.InvalidCredentialsException;

/**
 * This interface has services for User
 * @author Arya P Menon
 * @version 1.0
 */

public interface UserService {
	
	User saveUser(User user) throws EmailAlreadyExistsException;
	
	User updateUser(User user);
	
	User viewProfile(String email);
	
	Address addAddress(Address address, String email);
	
	CardDetails addCard(CardDetails card, String email);
	
	User login(Login login) throws InvalidCredentialsException;
	
	
	
	
	
	
}

File: buybeats-react-app/src/Services/UserService.js
import axios from 'axios'

const User_REST_API_URL="http://localhost:8870/users";

class UserService{

    // login(){
    //   return axios.post(User_REST_API_URL+"/login");
    // }
    addAddress(email){
        return axios.get(User_REST_API_URL+"/addAddress/"+email);
    }
}

export default new UserService()